Add UI for your vaults

adam5909099 opened the pull request:
# Task:
https://github.com/opynfinance/squeeth-monorepo/issues/186
![image](https://user-images.githubusercontent.com/59410529/156642680-ab88d3c0-f0db-47f3-a340-8eafcd6b8fea.png)


## Description
Add your vaults section which shows list of user's vaults whose collateral amount is greater than 0

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested
Go to positions and check if clicking vault in your vaults section redirect to vault detail.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

KMKoushik commented:
@adam5909099 There's an issue in deployment. Can you check?

KMKoushik commented on packages/frontend/src/components/LabelWithTooltip.test.tsx:
```diff
@@ -0,0 +1,23 @@
+import { render, screen } from '@testing-library/react'
+import LabelWithTooltip from './LabelWithTooltip'
+
+describe('LabelWithTooltip', () => {
```
Niceee!

adam5909099 commented:
@KMKoushik I tried checking the deployment issue but I don't have access to the deployment log.
CC @alexisgauba

KMKoushik commented:
@yichiehliu If you have any comments on tests please post it here

alexisgauba commented on packages/frontend/src/components/Trade/YourVaults.test.tsx:
```diff
@@ -0,0 +1,50 @@
+import YourVaults from './YourVaults'
+import * as useYourVaults from '../../hooks/useYourVaults'
+
+describe('YourVaults', () => {
```
love the testing!!

alexisgauba commented on packages/frontend/src/components/Trade/YourVaults.tsx:
```diff
@@ -0,0 +1,53 @@
+  return (
+    <>
+      {vaults?.map((vault, index) => (
+        <Link key={vault.id} href={`/vault/${vault.id}`} passHref>
```
Can we add labels in line instead of in the tooltip?

<img width="908" alt="Screen Shot 2022-03-03 at 8 35 36 PM" src="https://user-images.githubusercontent.com/13340486/156700571-440cf203-a854-4ce5-b13d-2369db925757.png">

yichiehliu commented:
it would be great if the commit msgs in the future PRs can follow the contribution guide [here](https://github.com/opynfinance/squeeth-monorepo/blob/main/packages/frontend/CONTRIBUTING.md)

yichiehliu commented:
> @yichiehliu If you have any comments on tests please post it here

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/26922518/156823101-0d2ba1d9-ca87-423c-a2f9-e602e2a98b6b.png">
I ran into a console error when I ran all of the tests, is this expected behavior?

yichiehliu commented on yarn.lock:
```diff
@@ -2,6 +2,13 @@
 # yarn lockfile v1
```
Do we commit this file?

yichiehliu replied:
Thinking that we dont need to commit yarn.lock file? cc @KMKoushik

KMKoushik replied:
We need to commit this file!

adam5909099 commented:
@alexisgauba I updated as you mentioned. I removed the tooltip because it's unnecessary
![image](https://user-images.githubusercontent.com/59410529/156863450-7820a043-7229-4d00-a17d-0f8aa721e85a.png)
@yichiehliu The problem was Material UI 4 Box component doesn't receive ref. It's resolved by wrapping it inside an HTML tag.

alexisgauba commented:
> https://continuouscall-git-186-user-vaults-disappear-opynfinance.vercel.app

Ah oh I definitely didn't mean the colors I had in my screenshot, sorry if that was confusing! I just meant that we should have labels ahah

In terms of styling would be great to keep the labels consistent with the rest of the labels on the screen, like in the red box below the labels are usually the same font, font-size, and color as the numbers at the moment

<img width="871" alt="Screen Shot 2022-03-04 at 7 33 56 PM" src="https://user-images.githubusercontent.com/13340486/156866183-fff04650-6dd2-4546-a25d-260595d4c06c.png">

alexisgauba reviewed (approved):
lgtm!
